runtime(filetype): Added suppprt for kitty

OXY2DEV opened the pull request:
This PR adds support for the `kitty` filetype(e.g. `~/.config/kitty/*.conf`, see details [here](https://sw.kovidgoyal.net/kitty/invocation/#cmdoption-kitty-config).

I have,

- [X] Updated the `runtime/filetype.vim` file with the appropriate `autocmd`.

>[!NOTE]
> As this filetype relies on `environment variables`(`$KITTY_CONFIG_DIR`, `$XDG_CONFIG_HOME` etc.) I have simplified the file path pattern. I have also made it so that any `kitty.conf` file is detected as thios filetype(let me know if this is a concern).

- [X] Updated `src/testdir/test_filetype.vim`.
- [x] Added `syntax/kitty.vim`.

> I am not sure if this needs a new syntax file as it's kinda similar to `bash` and the `conf` syntax works fine.
> You can check how the file may look like [here](https://github.com/OXY2DEV/tree-sitter-kitty/blob/57e8608da8c23381887e2c954e7d70a86b964535/images/kitty-demo.png).

OXY2DEV committed:
runtime(filetype): Added suppprt for kitty

OXY2DEV committed:
runtime(syntax): Added basic syntax for kitty

zeertzjq commented on src/testdir/test_filetype.vim:
```diff
@@ -419,6 +419,7 @@ def s:GetFilenameChecks(): dict<list<string>>
     jsp: ['file.jsp'],
     julia: ['file.jl'],
     just: ['justfile', 'Justfile', '.justfile', 'config.just'],
+    kitty: ['kitty.conf', '~/.config/kitty/colorscheme.conf'],
```
This should come between `kdl` and `kivy`.

OXY2DEV committed:
src(testdir): Updated test order

OXY2DEV committed:
runtime(syntax): Completed basic syntax for kitty

- Added support for `\` line continuation.
- Added `\r` support for line endings.

zeertzjq reviewed (commented):
```
Choose a good, descriptive name for your syntax file.  Use lowercase letters
and digits.  Don't make it too long, it is used in many places: The name of
the syntax file "name.vim", 'filetype', b:current_syntax and the start of each
syntax group (nameType, nameStatement, nameString, etc).

Start with a check for "b:current_syntax".  If it is defined, some other
syntax file, earlier in 'runtimepath' was already loaded: >

	if exists("b:current_syntax")
	  finish
	endif

Set "b:current_syntax" to the name of the syntax at the end.  Don't forget
that included files do this too, you might have to reset "b:current_syntax" if
you include two files.
```
- You need to set `b:current_syntax` at the end of the file.
- The syntax group names are inconsistent. They should all start with `kitty`.

OXY2DEV committed:
runtime(syntax): Updated syntax files for kitty

- [X] Use `kitty` prefix for syntax matches.
- [X] Set `b:current_syntax`.

jparise commented on runtime/syntax/kitty.vim:
```diff
@@ -0,0 +1,118 @@
+syn match kittyString /\S\+/ contains=Alpha contained
+syn match kittyNumber /[+\-*\/]\{0,1}[0-9.]\+/ contained
+syn match kittyAlpha /@[0-9.]\+/ contained
+syn match kittyColor /#[0-9a-fA-F]\{3,6}/ nextgroup=Alpha contained
```
Should this be `nextgroup=kittyAlpha`?

OXY2DEV replied:
Yes, because you can use colour names(e.g. `red@0.3`).

And I didn't want to make a huge regex pattern for it(or use a very long `syn keyword`). I was also unsure if it would have performance issues since there's quite a few colour names.

jparise replied:
Understood, but I think there's a typo: `contains=Alpha` versus `contains=kittyAlpha`?

OXY2DEV replied:
Yes, it's a typo. My bad. It should be `kittyAlpha`.

But, how do I push the changes? My new commit isn't showing up here anymore.

chrisbra replied:
Make a new PR with your changes please
